src/components/BenefitPlanTaskPreviewTable.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable react/no-array-index-key */
+import React from 'react';
+
+import { injectIntl } from 'react-intl';
+import { withTheme, withStyles } from '@material-ui/core/styles';
+import {
+  Table, TableBody, TableCell, TableContainer, TableHead, TableRow,
+} from '@material-ui/core';
+import {
+  formatMessage, PublishedComponent,
+} from '@openimis/fe-core';
+
+const styles = (theme) => ({
+  table: theme.table,
+  tableTitle: theme.table.title,
+  tableHeader: theme.table.header,
+  tableRow: theme.table.row,
+  tableLockedRow: theme.table.lockedRow,
+  tableLockedCell: theme.table.lockedCell,
+  tableHighlightedRow: theme.table.highlightedRow,
+  tableHighlightedCell: theme.table.highlightedCell,
+  tableHighlightedAltRow: theme.table.highlightedAltRow,
+  tableHighlightedAltCell: theme.table.highlightedAltCell,
+  tableDisabledRow: theme.table.disabledRow,
+  tableDisabledCell: theme.table.disabledCell,
+  tableFooter: theme.table.footer,
+  pager: theme.table.pager,
+  left: {
+    textAlign: 'left',
+  },
+  right: {
+    textAlign: 'right',
+  },
+  center: {
+    textAlign: 'center',
+  },
+  clickable: {
+    cursor: 'pointer',
+  },
+  loader: {
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+    background: 'rgba(0, 0, 0, 0.12)',
+  },
+});
+
+function BenefitPlanTaskPreviewTable({ intl, classes, previewItem }) {
+  const headers = () => [
+    'benefitPlan.code',
+    'benefitPlan.name',
+    'benefitPlan.type',
+    'benefitPlan.dateValidFrom',
+    'benefitPlan.dateValidTo',
+    'benefitPlan.maxBeneficiaries',
+  ];
+
+  const itemFormatters = () => [
+    (benefitPlan) => benefitPlan?.code,
+    (benefitPlan) => benefitPlan?.name,
+    (benefitPlan) => benefitPlan?.type,
+    (benefitPlan) => benefitPlan?.date_valid_from,
+    (benefitPlan) => benefitPlan?.date_valid_to,
+    (benefitPlan) => benefitPlan?.max_beneficiaries,
+  ];
+
+  const TASK_PREVIEW_FORMATTERS = itemFormatters();
+
+  return (
+    <TableContainer>
+      <Table size="small" className={classes.table}>
+        <TableHead>
+          <TableRow>
+            {headers().map((column) => (
+              <TableCell>{formatMessage(intl, 'socialProtection', column)}</TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {/* <ProgressOrError progress={fetchingTasks} error={errorTasks} /> */}
```

Comment:
It this commented line on purpose? Do we need this line? @jdolkowski . If no, please remove it.

Reply:
I think we can get _fetchingTasks_ and _errorTasks_ from the Redux store using _useSelector_ and then just keep this _ProgresOrError_ component.

Reply:
done